napari/layers/shapes/_accelerated_triangulate.py

```diff
@@ -0,0 +1,318 @@
+
+    normals = np.empty_like(path)
+    vec_len_arr = np.empty((len(path) - 1), dtype=np.float32)
+    for i in range(1, len(path)):
+        vec_diff = path[i] - path[i - 1]
+        vec_len_arr[i - 1] = np.sqrt(vec_diff[0] ** 2 + vec_diff[1] ** 2)
+        normals[i - 1] = vec_diff / vec_len_arr[i - 1]
+
+    if closed:
+        vec_diff = path[0] - path[-1]
+        vec_len_arr[-1] = np.sqrt(vec_diff[0] ** 2 + vec_diff[1] ** 2)
+        normals[-1] = vec_diff / vec_len_arr[-1]
+
+    point_count = _calc_output_size(normals, closed, cos_limit, bevel)
+
+    centers = np.empty((point_count, 2), dtype=np.float32)
+    offsets = np.empty((point_count, 2), dtype=np.float32)
+    triangles = np.empty((point_count - 2, 3), dtype=np.int32)
+
+    if closed:
+        j = _set_centers_and_offsets(
+            centers,
+            offsets,
+            triangles,
+            path[0],
+            normals[-1],
+            normals[0],
+            vec_len_arr[-1],
+            vec_len_arr[0],
+            0,
+            cos_limit,
+            bevel,
+        )
+    else:
+        centers[0] = path[0]
+        centers[1] = path[0]
+        offsets[0, 0] = normals[0][1] * 0.5
+        offsets[0, 1] = -normals[0][0] * 0.5
+        offsets[1] = -offsets[0]
+        triangles[0] = [0, 1, 2]
+        triangles[1] = [1, 2, 3]
+        j = 2
+    for i in range(1, len(normals) - 1):
+        j += _set_centers_and_offsets(
+            centers,
+            offsets,
+            triangles,
+            path[i],
+            normals[i - 1],
+            normals[i],
+            vec_len_arr[i - 1],
+            vec_len_arr[i],
+            j,
+            cos_limit,
+            bevel,
+        )
+    if closed:
+        j += _set_centers_and_offsets(
+            centers,
+            offsets,
+            triangles,
+            path[-1],
+            normals[-2],
+            normals[-1],
+            vec_len_arr[-2],
+            vec_len_arr[-1],
+            j,
+            cos_limit,
+            bevel,
+        )
+        centers[j] = centers[0]
+        centers[j + 1] = centers[1]
+        offsets[j] = offsets[0]
+        offsets[j + 1] = offsets[1]
+    else:
+        centers[j] = path[-1]
+        centers[j + 1] = path[-1]
+        offsets[j, 0] = normals[-2][1] * 0.5
+        offsets[j, 1] = -normals[-2][0] * 0.5
+        offsets[j + 1] = -offsets[j]
+
+    _fix_triangle_orientation(triangles, centers, offsets)
+
+    return centers, offsets, triangles
+
+
+@njit(cache=True)
+def remove_path_duplicates(path: np.ndarray, closed: bool) -> np.ndarray:
+    """Remove duplicates from a path.
+
+    Parameters
+    ----------
+    path : np.ndarray
+        Nx2 or Nx3 array of central coordinates of path to be deduplicated
+
+    Returns
+    -------
+    np.ndarray
+        Nx2 or Nx3 array of central coordinates of deduplicated path
+    """
+    if len(path) <= 2:
+        # part of ugly hack to keep lasso tools working
```

Comment:
Maybe I missed something, but if there will be no remarks changing my understanding I will open such Issue.

Reply:
@Czaki Let's open an issue and refer to it in the comment. :sunny:

Reply:
@Czaki can you explain what the issue is exactly? My interpretation:

> when the lasso tool "drops" a new point in a polygon being drawn, momentarily, the last to points in the polygon being drawn are identical. This happens also at the very beginning of drawing, when the first point is added to the polygon and the second point (which follows the mouse/pen tip) is also added immediately. Because of this, if we only have two points, we ignore it if they are duplicates.